### PR TITLE
[plugin.video.haystacktv] 1.0.1

### PR DESCRIPTION
**Pull request by Lunatixz**
### Checklist:

- [x] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-plugins/blob/master/CONTRIBUTING.md) document
- [x] Each add-on submission should be a single commit with using the following style: [plugin.video.foo] v1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practice but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-plugins/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.

### CODE REVIEW

**Comment by Lunatixz** [comment]
Travis `PROBLEM: Icon should have either 256x256 or 512x512 but it has 519x516` is a false positive. I double checked the icons dimensions and they are 512x512

File: plugin.video.haystacktv/resources/lib/haystack.py [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+        params=sys.argv[2]
+        cleanedparams=params.replace('?','')
+        if (params[len(params)-1]=='/'): params=params[0:len(params)-2]
+        pairsofparams=cleanedparams.split('&')
+        param={}
+        for i in range(len(pairsofparams)):
+            splitparams={}
+            splitparams=pairsofparams[i].split('=')
+            if (len(splitparams))==2: param[splitparams[0]]=splitparams[1]
+    return param
```
Comment (romanvm) [inline plugin.video.haystacktv/resources/lib/haystack.py]: The following one-liner is enough for simple query strings (without repeating keys):
```python
param = dict(urlparse.parse_qsl(sys.argv[2][1:]))
```

Comment (Lunatixz) [reply plugin.video.haystacktv/resources/lib/haystack.py]: Thanks, I'll look into the change.

Comment (Lunatixz) [reply plugin.video.haystacktv/resources/lib/haystack.py]: Accidentally closed the branch when updating recommend changes... I'll rebase.

**Pull request merged**